### ISSUE TITLE
Mutation is causing a THROWN error to propagate.. but not even sure WHY its throwing this error..

### ISSUE THREAD

**Issue by sjmcdowall** [zino-app/graphql-flutter#131]
**Describe the bug**
I am getting an (unexpected) Thrown Error in a Query Mutation call ... however, what appears to be the same mutation works in GraphiQL..

The relevant error stack is as follows:

E/flutter (15507): [ERROR:flutter/shell/common/shell.cc(181)] Dart Error: Unhandled exception:
E/flutter (15507): NoSuchMethodError: Class 'JsonUnsupportedObjectError' has no instance getter 'message'.
E/flutter (15507): Receiver: Instance of 'JsonUnsupportedObjectError'
E/flutter (15507): Tried calling: message
E/flutter (15507): #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
E/flutter (15507): #1      QueryManager.fetchQuery (package:graphql_flutter/src/core/query_manager.dart:142:24)
E/flutter (15507): <asynchronous suspension>
E/flutter (15507): #2      QueryManager.mutate (package:graphql_flutter/src/core/query_manager.dart:59:12)
E/flutter (15507): #3      GraphQLClient.mutate (package:graphql_flutter/src/graphql_client.dart:50:25)
E/flutter (15507): #4      _EventMessageEditScreenState._addEventMessage (file:///Users/sjm/work/mm/mms4/mobile/lib/screens/events/eventMessageEditScreen.dart:224:50)
E/flutter (15507): <asynchronous suspension>
E/flutter (15507): #5      _EventMessageEditScreenState._submitForm (file:///Users/sjm/work/mm/mms4/mobile/lib/screens/events/eventMessageEditScreen.dart:276:26)
E/flutter (15507): <asynchronous suspension>

**To Reproduce**
The GraphQL mutation is:

```
String createEventMessage = """
  mutation createEventMessage(\$customerId: Int!, \$eventId: Int!, \$memberId: Int!, 
  \$nameFirst: String!, \$nameMiddle: String!, \$nameLast: String!,
  \$subject: String!, \$content: String!, \$messageClass: Int!) {
  createMessage(
    input: {
      message: {
        customerId: \$customerId,
        eventId: \$eventId,
        memberId: \$memberId,
        messageSubject: \$subject,
        content: \$content,
        nameFirst: \$nameFirst,
        nameMiddle: \$nameMiddle,
        nameLast: \$nameLast,
        messageClass: \$messageClass,
        miimeType: "text/plain",
        isSticky: 0
	    }
    }
    ) {
      message {
        messageId
        createdTime
      }
    }
	}

"""
    .replaceAll('\n', ' ');
```

With the call being:

```
    QueryResult result = await globals.gqlClient.mutate(
      MutationOptions(
        document: createEventMessage,
        variables: {
          'customerId': globals.user.customerId,
          'eventId': widget.eventId,
          'memberId': globals.user.memberId,
          'messageClass': MessageClass.MESSAGE_CLASS_USER,
          'messageSubject': subject,
          'content': content,
          'nameFirst': globals.user.nameFirst,
          'nameMiddle': globals.user.nameMiddle,
          'nameLast': globals.user.nameLast,
        },
        fetchPolicy: FetchPolicy.networkOnly,
        context: <String, dynamic>{
          'headers': <String, String>{
            'Authorization': 'Bearer ${globals.user.token}',
            'X-MM-Organization': '${globals.user.customerId}',
          }
        },
      ),
    );
```

**Expected behavior**
Well I am hoping the actual mutation works -- but at a minimum I would have expected it to catch the error and then return that in the result.errors aspect .. 

However, I am not sure why the error is thrown in the first place?  Here is the working GraphiQL screen shot

<img width="1739" alt="screen shot 2018-11-01 at 1 40 27 pm" src="https://user-images.githubusercontent.com/2034925/47868873-c3f52780-dddb-11e8-9243-93358b9f4cbd.png">

**Desktop (please complete the following information):**
 Mac OS 10.14

**Smartphone (please complete the following information):**
Android Emulator SDK 27

**Comment by sjmcdowall** [comment]
It appears the error is being thrown here - which is hiding some underlying error .. 

[Error Line](https://github.com/zino-app/graphql-flutter/blob/next/lib/src/core/query_manager.dart#L142)

Thanks to my co-worker Scott for tracking this down... one step closer..

**Comment by venkatd** [comment]
@sjmcdowall I'm running into issues around having this library swallow errors as well. I think the problem is that the catch statement swallows *all* errors instead of catching just the specific errors that need to be transformed into graphql errors.

As far as I can tell, Dart has the ability to catch errors explicitly https://stackoverflow.com/a/20803316/67655

If an exception does not fall under a graphql specific error, it probably should not to be caught.

**Comment by kashyapnayan** [comment]
Getting the same error can you please elaborate how can i get rid of this error @sjmcdowall

But the thing is i am not using mutation, i am using Query because  i want to fetch the data from graphql.
In that query when i am passing the variable default value like any integer value it is working fine but when i am trying to get that variable value from SharedPreferences and trying passing it to query it is giving me this error.

Can you please help.
Thanks in advance

**Comment by sjmcdowall** [comment]
Your problem -- I think -- is that getUserId is an async function (and rightly so) -- but you are NOT calling it async in superInit (because you can not) .. 

You need to wait until the userId is actually set .. and that "var prefs" is ugly too :) 

You need to handle in your build() the condition where userId is null (and return an "In progress" sort of thing) and only redner the Query() widget if userId is set .. 

and then (maybe in your getUserid -- do something like

if (mounted) {
 setState() {};
}

to cause a new build() .. now with userId set .. (i'd set it right there in the program) .. 

You also do not want your Link and Valuenotifiier stuff like that in ANY build() routine .. that can get 100's / second (assume it can) .. and those should only be done ONCE in the app launch .. 

And don't use "var" for anything -- remove that from your Javascript mind.. 

Grab the google analyze_options.yaml  anduse VS Code .. and make sure to clean up all the errors it finds and keep it clean .. 

The last items are just freebie tips :) 

But basically you have a fundamental flaw in your knowledge of Flutter flow .. like we all do when we start :)

**Comment by micimize** [comment]
If I understand correctly, the main issue is error swallowing which will be solved by #193.
@sjmcdowall were you ever able to figure out the underlying cause

**Comment by micimize** [comment]
closing this as the error swallowing should be solved by #193, and the other issue will need more details
